bugshaker/src/main/java/com/github/stkent/bugshaker/MainActivity.java

```diff
@@ -0,0 +1,279 @@
+			mediumBtn.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setErase(true);
+					drawView.setBrushSize(mediumEraser);
+					brushDialog.dismiss();
+				}
+			});
+
+			ImageButton largeBtn = (ImageButton) brushDialog.findViewById(R.id.large_brush);
+			largeBtn.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setErase(true);
+					drawView.setBrushSize(largeEraser);
+					brushDialog.dismiss();
+				}
+			});
+			brushDialog.show();
+		}
+
+		else if (view.getId()==R.id.sendEmail){
+			saveDrawing();
+		}
+		else if (view.getId()==R.id.speechBox){
+
+			createTextEdit();
+		}
+
+
+
+	}
+
+	private void createTextEdit(){
+		//get x and y
+
+	}
+
+	public void paintClicked(View view){
+		//use chosen color
+		drawView.setBrushSize(drawView.getLastBrushSize());
+
+		if(view!=currPaint){
+			//update color
+			ImageButton imgView = (ImageButton)view;
+			String clickedButtonColor = view.getTag().toString();
+			drawView.setColor(clickedButtonColor);
+
+			imgView.setImageDrawable(getResources().getDrawable(R.drawable.paint_pressed));
+			currPaint.setImageDrawable(getResources().getDrawable(R.drawable.paint));
+			currPaint=(ImageButton)view;
+		}
+	 }
+
+	private Uri getImageUri(Context inContext, Bitmap inImage) {
+		ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+		inImage.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
+		String path = MediaStore.Images.Media.insertImage(inContext.getContentResolver(), inImage, "Title", null);
+		return Uri.parse(path);
+	}
+
+	private void saveDrawing() {
+		AlertDialog.Builder sendDialog = new AlertDialog.Builder(this);
+
+		sendDialog.setTitle(getApplicationContext().getString(R.string.send_annotated_screenshot));
+		sendDialog.setMessage(getApplicationContext().getString(R.string.attach_annotated_screenshot_to_email));
+		sendDialog.setPositiveButton(getApplicationContext().getString(R.string.yes), new DialogInterface.OnClickListener(){
+			public void onClick(DialogInterface dialog, int which){
+				drawView.setDrawingCacheEnabled(true);
+
+				//Saves image in phone Gallery
+				String imgSaved = MediaStore.Images.Media.insertImage(
+					getContentResolver(), drawView.getDrawingCache(),
+					ScreenshotUtil.getImageFileName(), ScreenshotUtil.getImageDescription());
+				Bitmap screenshotBitmap = drawView.getDrawingCache();
+
+				Uri bitmapUri = getImageUri(getApplicationContext(), screenshotBitmap);
+				System.out.println(bitmapUri.getPath());
+
+				if (imgSaved != null) {
+					Toast savedToast = Toast.makeText(getApplicationContext(),
+						"Drawing saved to Gallery!", Toast.LENGTH_SHORT);
+
+
+					File log = LogcatUtil.saveLogcatToFile(getApplicationContext());
+					Uri logUri = Uri.fromFile(log);
+					feedbackEmailFlowManager.sendEmailWithScreenshot(activity, bitmapUri, logUri);
+
```

Comment:
this code should be in a commonplace as well. Maybe this class should not have anything to do with the Logcat